### PR TITLE
Feature1

### PR DESCRIPTION
**Pull request by ciarasugrue**

### CODE REVIEW

File: feature1.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    
+ALTER TABLE Employee
+ADD CONSTRAINT fk_team_id
+FOREIGN KEY (team_id)
+REFERENCES Team(team_id);
+
+ALTER TABLE SalesEmployee
+ADD CONSTRAINT fk_employee_id
+FOREIGN KEY (employee_id)
+REFERENCES Employee(employee_id);
```
Comment (shaunganley) [inline feature1.sql:33]: can you add the FK when creating the table rather than here

File: feature1.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+-- Insert fake data into the SalesEmployee table
+INSERT INTO SalesEmployee (employee_id, commission_rate)
+VALUES
+    (5, 0.10),
+    (8, 0.08);
+
+-- We are assuming that we can restrict to HR employees to add a Sales employee using Java/web dev technologies.
+-- Query to find HR employees:
+SELECT * FROM Employee
+WHERE team_id = 2;
```
Comment (shaunganley) [inline feature1.sql:60]: the first feature asks to see a sales employee so you should select from sales employee here and join with the employee table. also; don't use select *, specify column names

File: feature1.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+-- Insert fake data into the Employee table
+INSERT INTO Employee (employee_name, salary, bank_account_number, national_insurance_number, team_id)
+VALUES
+    ('John Doe', 60000, '12345678', 'AB123456C', 1),
+    ('Jane Smith', 55000, '23456789', 'CD234567D', 2),
+    ('David Johnson', 70000, '34567890', 'EF345678E', 2),
+    ('Emily Davis', 48000, '45678901', 'GH456789F', 1);
+    
+select * from Employee;
```
Comment (shaunganley) [inline feature1.sql:49]: don't use select *, specify column names

File: feature1.sql [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    employee_name VARCHAR(255),
+    salary MEDIUMINT,
+    bank_account_number CHAR(8),
+    national_insurance_number CHAR(9),
+    team_id MEDIUMINT
+	);
+    
+CREATE TABLE SalesEmployee (
+	sales_employee_id MEDIUMINT PRIMARY KEY AUTO_INCREMENT,
+    employee_id MEDIUMINT,
```
Comment (shaunganley) [inline feature1.sql:21]: having this as a FK makes this a 1:many relationship, is that what we want to have? think about what sort of test data you may be able to enter in the employee_id column

File: feature1.sql [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+CREATE TABLE Team (
+	team_id MEDIUMINT PRIMARY KEY AUTO_INCREMENT,
+    team_type VARCHAR(255)
+    );
+    
+CREATE TABLE Employee (
+	employee_id MEDIUMINT PRIMARY KEY AUTO_INCREMENT,
+    employee_name VARCHAR(255),
+    salary MEDIUMINT,
```
Comment (shaunganley) [inline feature1.sql:13]: should this be of type MEDIUMINT?

File: ciarascript.sql
Comment (shaunganley) [inline ciarascript.sql:1]: I think you can delete this file

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: feature1.sql (hunk 3 above)
Comment (ciarasugrue) [reply feature1.sql:49]: this query was just to check table contents.

**Review by shaunganley — APPROVED** [review]

**Pull request merged**